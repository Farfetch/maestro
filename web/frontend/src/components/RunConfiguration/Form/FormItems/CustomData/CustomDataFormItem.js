import { Form } from "antd";
import React from "react";

import UploadDragger from "../../../../form/UploadDragger";

const normFile = (e) => {
  if (Array.isArray(e)) {
    return e;
  }
  return e && e.fileList;
};
const CustomDataFormItem = () => (
  <>
    {/* TODO: add validator to check file extention */}
    <Form.Item label="">
      <Form.Item
        name="customData"
        valuePropName="fileList"
        getValueFromEvent={normFile}
        noStyle
      >
        <UploadDragger
          name="customData"
          multiple={true}
          description="Select list of CSV files to be used in test"
        />
      </Form.Item>
    </Form.Item>
  </>
);

export default CustomDataFormItem;
