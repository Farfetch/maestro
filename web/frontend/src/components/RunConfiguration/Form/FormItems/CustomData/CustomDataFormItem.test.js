import { cleanup, render, waitFor } from "@testing-library/react";
import { Form } from "antd";
import React from "react";

import CustomDataFormItem from "./CustomDataFormItem";

afterEach(cleanup);

describe("components/RunConfiguration/Form/CustomDataFormItem", () => {
  test(`should render CustomDataFormItem component`, async () => {
    const { container } = render(
      <Form>
        <CustomDataFormItem />
      </Form>
    );

    await waitFor(() => {
      expect(container.outerHTML).toMatchSnapshot();
    });
  });
});
