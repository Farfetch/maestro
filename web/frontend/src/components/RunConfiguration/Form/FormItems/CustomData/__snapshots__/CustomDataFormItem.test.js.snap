// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RunConfiguration/Form/CustomDataFormItem should render CustomDataFormItem component 1`] = `"<div><form class=\\"ant-form ant-form-horizontal css-dev-only-do-not-override-nnuwmp\\"><div class=\\"ant-form-item css-dev-only-do-not-override-nnuwmp\\"><div class=\\"ant-row ant-form-item-row css-dev-only-do-not-override-nnuwmp\\"><div class=\\"ant-col ant-form-item-control css-dev-only-do-not-override-nnuwmp\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><span class=\\"ant-upload-wrapper css-dev-only-do-not-override-nnuwmp\\"><div class=\\"css-dev-only-do-not-override-nnuwmp ant-upload ant-upload-drag\\" style=\\"margin-bottom: 0px;\\"><span tabindex=\\"0\\" class=\\"ant-upload ant-upload-btn\\" role=\\"button\\"><input type=\\"file\\" style=\\"display: none;\\" accept=\\"\\" multiple=\\"\\"><div class=\\"ant-upload-drag-container\\"><div style=\\"display: flex;\\"><div style=\\"width: 60px;\\"><span role=\\"img\\" aria-label=\\"inbox\\" style=\\"color: rgb(64, 169, 255); font-size: 40px;\\" class=\\"anticon anticon-inbox\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"inbox\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M885.2 446.3l-.2-.8-112.2-285.1c-5-16.1-19.9-27.2-36.8-27.2H281.2c-17 0-32.1 11.3-36.9 27.6L139.4 443l-.3.7-.2.8c-1.3 4.9-1.7 9.9-1 14.8-.1 1.6-.2 3.2-.2 4.8V830a60.9 60.9 0 0060.8 60.8h627.2c33.5 0 60.8-27.3 60.9-60.8V464.1c0-1.3 0-2.6-.1-3.7.4-4.9 0-9.6-1.3-14.1zm-295.8-43l-.3 15.7c-.8 44.9-31.8 75.1-77.1 75.1-22.1 0-41.1-7.1-54.8-20.6S436 441.2 435.6 419l-.3-15.7H229.5L309 210h399.2l81.7 193.3H589.4zm-375 76.8h157.3c24.3 57.1 76 90.8 140.4 90.8 33.7 0 65-9.4 90.3-27.2 22.2-15.6 39.5-37.4 50.7-63.6h156.5V814H214.4V480.1z\\"></path></svg></span></div><span class=\\"ant-typography css-dev-only-do-not-override-nnuwmp\\" style=\\"line-height: 40px;\\">Select list of CSV files to be used in test</span></div></div></span></div><div class=\\"ant-upload-list ant-upload-list-text\\"></div></span></div></div></div></div></div></form></div>"`;
